homeassistant/components/opensky/coordinator.py

```diff
@@ -0,0 +1,116 @@
+"""DataUpdateCoordinator for the OpenSky integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from python_opensky import OpenSky, OpenSkyError, StateVector
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_ALTITUDE,
+    ATTR_CALLSIGN,
+    ATTR_ICAO24,
+    ATTR_SENSOR,
+    CONF_ALTITUDE,
+    DEFAULT_ALTITUDE,
+    DOMAIN,
+    EVENT_OPENSKY_ENTRY,
+    EVENT_OPENSKY_EXIT,
+    LOGGER,
+)
+
+
+class OpenSkyDataUpdateCoordinator(DataUpdateCoordinator[int]):
+    """An OpenSky Data Update Coordinator."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, opensky: OpenSky) -> None:
+        """Initialize the OpenSky data coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            # OpenSky free user has 400 credits, with 4 credits per API call. 100/24 = ~4 requests per hour
+            update_interval=timedelta(minutes=15),
+        )
+        self._opensky = opensky
+        self._previously_tracked: set[str] | None = None
+        self._bounding_box = OpenSky.get_bounding_box(
+            self.config_entry.data[CONF_LATITUDE],
+            self.config_entry.data[CONF_LONGITUDE],
+            self.config_entry.options[CONF_RADIUS],
+        )
+        self._altitude = self.config_entry.options.get(CONF_ALTITUDE, DEFAULT_ALTITUDE)
+
+    async def _async_update_data(self) -> int:
+        try:
+            response = await self._opensky.get_states(bounding_box=self._bounding_box)
+        except OpenSkyError as exc:
+            raise UpdateFailed from exc
+        currently_tracked = set()
+        flight_metadata: dict[str, StateVector] = {}
+        for flight in response.states:
+            if not flight.callsign:
+                continue
+            callsign = flight.callsign.strip()
+            if callsign != "":
+                flight_metadata[callsign] = flight
+            else:
+                continue
+            if (
+                flight.longitude is None
+                or flight.latitude is None
+                or flight.on_ground
+                or flight.barometric_altitude is None
+            ):
+                continue
+            altitude = flight.barometric_altitude
+            if altitude > self._altitude and self._altitude != 0:
```

Comment:
Just double-checking...

In the docs, it is written:
> The minimum height in meters above which the planes are tracked. No input (or 0) means all planes are tracked in the area.

And the code does the opposite. (It was already like this before this PR)

Reply:
Euh, it now checks if the altitude of the plain is bigger than the set altitude (so if they fly above it) and if the set altitude is 0, it will not enter this if statement but add it to the currently tracked list

Reply:
Docs were wrong updated and will be fixed with https://github.com/home-assistant/home-assistant.io/pull/28533